src/state/middlewares/system/controller.js

```diff
@@ -61,164 +101,211 @@ const formatAgentMessage = ( author_type, author_id, session_id, { id, timestamp
 	type,
 	meta,
 	source
-} )
+} );
 
-export default ( middlewares ) => store => {
-	const cache = { operator: new ChatLog( { maxMessages: 20 } ), customer: new ChatLog( { maxMessages: 20 } ) }
+const defaultChatLog = () => new ChatLog( { maxMessages: 20 } );
 
-	const runMiddleware = ( ... args ) => run( middlewares )( ... args )
+/**
+ * Creates controlled middleware
+ *
+ * @param { Function } messageFilter - list of message filters to be applied to every message
+ * @returns { Function } redux middleware
+ */
+export default ( messageFilter, chatLogFactory = defaultChatLog ) => store => {
+	const cache = { operator: chatLogFactory( 'operator' ), customer: chatLogFactory( 'customer' ) };
```

Comment:
When `chatLogFactory` uses `defaultChatLog`, the parameters `'operator'` and `'customer'` are not used. Is this intentional, assuming that some other consumer of `happychat-service` will want to use those values in their own log factory function?